### PR TITLE
Add count to execution repo interfaces

### PR DESCRIPTION
**Pull request by andrewwdye**
# TL;DR
Add a `Count` interface and implementation to `ExecutionRepo`, `NodeExecutionRepo`, and `TaskExecutionRepo`. This allows counting executions matching a filter set without needing to materialize the list.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Workflow, node, and task execution states are stored in a GORM DB. This change adds a `Count` interface to `ExecutionRepo`, `NodeExecutionRepo`, and `TaskExecutionRepo` to query the count of executions efficiently (instead of materializing the list and summing). The interface accepts `InlineFilters` and `MapFilters`, similar to the `List` interfaces and the `ListResourceInput` param.

Relevant unit tests added.

This change also 
* Runs `make goimports`, which led to updates in a number of unrelated files
* Refactors existing repo mocks for consistency

## Tracking Issue
NA

## Follow-up issue
NA

### CODE REVIEW

File: pkg/manager/impl/execution_manager_test.go [hunk 1]
```diff
@@ -2818,7 +2818,7 @@ func TestTerminateExecution(t *testing.T) {
 		}, unmarshaledClosure.GetAbortMetadata()))
 		return nil
 	}
-	repository.ExecutionRepo().(*repositoryMocks.MockExecutionRepo).SetUpdateExecutionCallback(updateExecutionFunc)
+	repository.ExecutionRepo().(*repositoryMocks.MockExecutionRepo).SetUpdateCallback(updateExecutionFunc)
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:2821]: thanks for simplifying these, we should really move to using mockery so these don't have to be handrolled (https://github.com/flyteorg/flyte/issues/149)

File: pkg/repositories/gormimpl/execution_repo.go [hunk 2]
```diff
@@ -110,6 +110,23 @@ func (r *ExecutionRepo) List(ctx context.Context, input interfaces.ListResourceI
 	}, nil
 }
 
+func (r *ExecutionRepo) Count(ctx context.Context, input interfaces.CountResourceInput) (int64, error) {
+	var err error
+	tx := r.db.Model(&models.Execution{})
+	tx, err = applyScopedFilters(tx, input.InlineFilters, input.MapFilters)
```
Comment (katrogan) [inline pkg/repositories/gormimpl/execution_repo.go:116]: for completeness can we add the calls to [JoinTableEntities](https://github.com/flyteorg/flyteadmin/blob/master/pkg/repositories/gormimpl/execution_repo.go#L78,L90) so we can support cross table filters

File: pkg/server/service.go [hunk 3]
```diff
@@ -305,7 +305,13 @@ func serveGatewayInsecure(ctx context.Context, pluginRegistry *plugins.Registry,
 		handler = httpServer
 	}
 
-	err = http.ListenAndServe(cfg.GetHostAddress(), handler)
+	server := &http.Server{
+		Addr:              cfg.GetHostAddress(),
+		Handler:           handler,
+		ReadHeaderTimeout: time.Duration(cfg.ReadHeaderTimeoutSeconds) * time.Second,
```
Comment (katrogan) [inline pkg/server/service.go:311]: thank you for adding this!

**Review by katrogan — COMMENTED** [review]
looks great! thanks for adding all the clean-up too

**Comment by andrewwdye** [comment]
Fyi, created https://github.com/flyteorg/flyte/issues/2814 for previously failing Check Go Generate. Looks like an inconsistency between go 1.18 and 1.19.

**Review by katrogan — APPROVED** [review]
